mitmproxy/tools/console/flowdetailview.py

```diff
@@ -29,10 +29,18 @@ def flowdetails(state, flow: http.HTTPFlow):
 
     if sc is not None:
         text.append(urwid.Text([("head", "Server Connection:")]))
-        parts = [
-            ["Address", "{}:{}".format(sc.address[0], sc.address[1])],
-            ["Resolved Address", "{}:{}".format(sc.ip_address[0], sc.ip_address[1])],
-        ]
+        if sc.address:
+            parts = [
+                ["Address", "{}:{}".format(sc.address[0], sc.address[1])],
+            ]
+        else:
+            parts = [
+                ["Address", "Couldn't Resolve"],
+            ]
+        if sc.ip_address:
+            parts.append(["Resolved Address", "{}:{}".format(sc.ip_address[0], sc.ip_address[1])])
+        else:
+            parts.append(["Resolved Address", "Couldn't Resolve"])
```

Comment:
This is a bit tricky here because `.ip_address` is only set after successful connection. It could very well be that we resolved the hostname successfully, but then failed to connect. In this case, we should not display `"Couldn't Resolve"`, because that is not what is happening. I suggest to just not display "Resolved Address" when `ip_address` is None.